client/js/controllers/editor.js

```diff
@@ -1036,6 +1038,130 @@
                     }
                 };
 
+                $scope.getNodes = function () {
+                    $http.get('/api/nodetypes/get')
+                    .success(function (data) {
+                        if (data) {
+                            for (var i in data) {
+                                for (var j in data[i]) {
+                                    nodenames.push(data[i][j].name);
+                                    inports.push(data[i][j].in_ports);
+                                    outports.push(data[i][j].out_ports);
+                                    if (data[i][j].options) {
+                                        members.push(data[i][j].options.members);
+                                        membernodes.push(data[i][j].name);
+                                    }
+                                }
+                            }
+                        }
+                    }).error(function (data) {
+                        console.log("Error getting nodetypes: " + data);
+                    });
+                };
+
+                $scope.getNodes();
+
+                var prettifyNodetype = function (nodetype) {
+                    var temp = nodetype.split('/');
+                    temp[0] = temp[0].toUpperCase();
+                    if (temp[1]) {
+                        temp[1] = temp[1].charAt(0).toUpperCase() + temp[1].slice(1);
+                        return temp[0] + ' ' + temp[1];
+                    }
+                };
+
+                var autoCompleter = {
+                    getCompletions: function(editor, session, pos, prefix, callback) {
+                        var curLine = session.getDocument().getLine(pos.row);
+                        var curTokens = curLine.slice(0, pos.column).split(/\s+/);
+                        var valTokens = curLine.slice(0, pos.column).split(':');
+                        if (valTokens[0]) {
+                            var nodeval = valTokens[0].lastIndexOf('(');
+                            if (nodeval > -1) {
+                                var valCmd = valTokens[0].slice(nodeval + 1);
```

Comment:
This variable valCmd is out of the scope.

This can provoke bugs that we don't want to when using the valCmd below without declaring it

Here for more info:
http://jshint.com/docs/options/#funcscope